### PR TITLE
Minimal creation function for AuthenticatorState

### PR DESCRIPTION
**Pull request by mystenmark**
This is to get the creation function for the authenticator state object out in the wild in advance of shipping the actual zklogin PR.

### CODE REVIEW

File: crates/sui-framework/packages/sui-framework/sources/authenticator_state.move [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    #[allow(unused_field)]
+    struct ActiveJwk has store, drop, copy {
+        jwk_id: JwkId,
+        jwk: JWK,
+        epoch: u64,
+    }
+
+    #[allow(unused_function)]
```
Comment (amnn) [inline crates/sui-framework/packages/sui-framework/sources/authenticator_state.move:61]: I think you either need to make this private and then you need the annotation, or you keep it public and you don't need the annotation.

**Review by joyqvq — APPROVED** [review]

File: crates/sui-framework/packages/sui-framework/sources/authenticator_state.move [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    /// Create and share the AuthenticatorState object. This function is call exactly once, when
+    /// the authenticator state object is first created.
+    /// Can only be called by genesis or change_epoch transactions.
+    public fun create(ctx: &TxContext) {
+        assert!(tx_context::sender(ctx) == @0x0, ENotSystemAddress);
+
+        let version = CurrentVersion;
```
Comment (amnn) [inline crates/sui-framework/packages/sui-framework/sources/authenticator_state.move:68]: nit: `UPPER_SNAKE_CASE` for this kind of constant in Move (conventionally).

**Review by amnn — APPROVED** [review]
LGTM, couple of nits!

File: crates/sui-framework/packages/sui-framework/sources/authenticator_state.move (hunk 1 above)
Comment (mystenmark) [reply crates/sui-framework/packages/sui-framework/sources/authenticator_state.move:61]: sorry, thought i did that. fixed.

Comment (lxfind) [reply crates/sui-framework/packages/sui-framework/sources/authenticator_state.move:61]: If the goal is to call it from Rust, then you could keep it private

Comment (mystenmark) [reply crates/sui-framework/packages/sui-framework/sources/authenticator_state.move:61]: yes, leaving it private

File: crates/sui-framework/packages/sui-framework/sources/authenticator_state.move (hunk 2 above)
Comment (mystenmark) [reply crates/sui-framework/packages/sui-framework/sources/authenticator_state.move]: done

**Pull request closed without merging**